src/redux/actions.js

```diff
@@ -0,0 +1,75 @@
+import axios from "axios";
+import Swal from "sweetalert2";
+import {
+  SIGNIN,
+  SIGNUP,
+  SIGNINPROCESSING,
+  SIGNUPPROCESSING,
+} from "./actionTypes.js";
+
+export const startSignin = (content) => {
+  return (dispatch) => {
+    dispatch(signinProcessing());
+    axios
```

Comment:
1) don't use dispatch - you can just use action creators
2) actions are just mutating of redux, don't mixin additional logic here (like api calls). do it separately

whatch my example project and rethink the way you build your project